misc/package/Data Files/MWSE/core/mcm/components/settings/PercentageSlider.lua

```diff
@@ -0,0 +1,44 @@
+--[[
+	PercentageSlider:
+		A slider that displays percentages as whole numbers, but stores the values as decimal numbers.
+		i.e., values get shown in the range [0,100], but are stored in the config as [0,1].
+]]--
+
+--- These types have annotations in the core\meta\ folder. Let's stop the warning spam here in the implementation.
+--- The warnings arise because each field set here is also 'set' in the annotations in the core\meta\ folder.
+--- @diagnostic disable: duplicate-set-field
+
+local Parent = require("mcm.components.settings.Slider")
+
+
+
+--- @class mwseMCMPercentageSlider : mwseMCMSlider
+local PercentageSlider = Parent:new()
+
+PercentageSlider.min = 0.0
+PercentageSlider.max = 1.0
+PercentageSlider.step = 0.01
+PercentageSlider.jump = 0.05
+
+function PercentageSlider:convertToWidgetValue(variableValue)
+	return (variableValue - self.min) * 10 ^ (2 + self.decimalPlaces)
+end
+
+function PercentageSlider:updateValueLabel()
+	local labelElement = self.elements.label
+
+	if string.find(self.label, "%s", 1, true) then
+		labelElement.text = string.format(self.label, self.variable.value)
+	else
+		local s = "%s: %i%%"
+		-- only include decimal places when we're supposed to
+		if self.decimalPlaces > 0 then
+			-- so sorry that anyone has to look at this
+			-- this will simplify to "%s: %.1f%%" (in the case where `decimalPlaces` == 1)
+			s = string.format("%%s: %%.%uf%%%%", self.decimalPlaces)
```

Comment:
The other option is to build the string by concatenation:
```lua
s = "%s: %." .. self.decimalPlaces .. "f%%"
```
Not that it's much more readable.

Reply:
that's another good idea and it really cuts down on the number of `%` signs which is always good. it's times like these where i reallly miss python fstrings